crates/bevy_app/src/app.rs

```diff
@@ -328,7 +328,15 @@ impl App {
     pub fn add_state<S: States>(&mut self) -> &mut Self {
         self.init_resource::<State<S>>();
         self.init_resource::<NextState<S>>();
-        self.add_systems(
+
+        let mut schedules = self.world.resource_mut::<Schedules>();
+
+        let Some(default_schedule) = schedules.get_mut(&*self.default_schedule_label) else {
```

Comment:
Won't this fail if you are currently running in the default schedule label?

Reply:
This is a method on `App`, so no schedule will be running.